portfolio/src/main/webapp/script.js

```diff
@@ -1,28 +1,29 @@
-// Copyright 2019 Google LLC
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     https://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+//this is the script js file to get sample data from server
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+function getMyName() {
+  fetch('/data').then(response => response.text()).then((name) => {
+    document.getElementById('name-container').innerText = name;
+  });
+}
+
+/** Fetches all facts */
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+function getMyFacts() {
+    fetch("/data").then(response => response.json()).then((facts) => {
+        console.log(facts);
+        const factsListElement = document.getElementById('liElement-container');
+        factsListElement.innerHTML = '';
+        for (let i = 0; i < facts.length; i++){
```

Comment:
given that the index isn't important, consider using a for/of loop: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

Reply:
I never thought that this would be an issue. Thanks for pointing it out.